src/templates/blog-list.js

```diff
@@ -1,55 +1,35 @@
-import React, { useEffect, useContext } from "react";
-
+import React from "react";
 import { graphql } from "gatsby";
 
-import { Pager } from '../components/pager';
-import { AppContext, BannerType } from '../config/context';
-
-import { Card } from '../components/card';
-
-import * as styles from "./blog-list.module.scss";
-
-const BlogList = ({ pageContext, data }) => {
-  const _ = require("lodash")    
-  const { edges } = data.allMdx;
-  const { setState } = useContext(AppContext);
+import { BlogList as BlogListComponent } from "../components/blog-list/blog-list"
 
-  useEffect(() => {
-    setState(BannerType.home);
-  }, []);
-
-  return (
-    <>
-      <div className={styles.container}>
-        {edges.map(({ node }) => <Card data={node} key={node.id} withCategory={true} />)}     
-      </div>
-      <Pager pageContext={pageContext}/>
-    </>
-  );
+const BlogList = (props) => {
+  return <BlogListComponent {...props} />;
```

Comment:
Is this only because you cannot import the GQL query elsewhere?

Reply:
Both the query and the component had the same issue and this helped me get around the component side of things 😬